test/functional/tools/filter_data_table.xml

```diff
@@ -0,0 +1,42 @@
+<tool id="filter_data_table" name="filter data table" version="0.1.0">
+    <description>Filter on datatable entries</description>
+    <command><![CDATA[
+       echo $dynamic_select > '$output'
+    ]]></command>
+    <inputs>
+        <param name="dynamic_select" type="select">
+            <!-- initialize options from file (contains all entries that could be added to the data table) -->
+            <options from_url="https://usegalaxy.org/api/genomes">
+                <column name="name" index="0"/>
+                <column name="value" index="0"/>
+                <!-- filter options that are already in the data table -->
+                <filter type="data_table"  column="1" table_name="test_fasta_indexes" data_table_column="1" keep="true"/>
```

Comment:
I have nightmares about the complexities here. Can we at least only allow named columns ?